[CI] Test files under src/ break typecheck build

## Problem

`pnpm build` (`tsc -p tsconfig.build.json --noEmit`) fails because `src/api/memory/deduplication.test.ts` imports from `tests/helpers/db.ts`, which is outside the `rootDir` (`src/`).

## Current Behavior

```
src/api/memory/deduplication.test.ts(9,51): error TS6059: File 'tests/helpers/db.ts' is not under 'rootDir' 'src'. 'rootDir' is expected to contain all source files.
```

CI `test` job fails on the Typecheck step. This affects **all branches including main**.

## Root Cause

`tsconfig.build.json` includes `src/**/*` but does not exclude test files (`*.test.ts`). When a test file under `src/` imports from `tests/`, TypeScript errors because `tests/` is outside the configured `rootDir`.

## Fix

Add `"src/**/*.test.ts"` to the `exclude` array in `tsconfig.build.json`:

```json
{
  "extends": "./tsconfig.json",
  "include": ["src/**/*"],
  "exclude": ["node_modules", "dist", "tests/**/*", "src/ui/**/*", "src/**/*.test.ts"]
}
```

## Context

- Discovered while working on #1166 (PR #1197)
- Fix included in PR #1197 to unblock CI
- Affects main branch CI (confirmed failing)
- Related to #859 (CI missing typecheck step)

--- a/tsconfig.build.json
+++ b/tsconfig.build.json
@@ -1,5 +1,5 @@
 {
   "extends": "./tsconfig.json",
   "include": ["src/**/*"],
-  "exclude": ["node_modules", "dist", "tests/**/*", "src/ui/**/*"]
+  "exclude": ["node_modules", "dist", "tests/**/*", "src/ui/**/*", "src/**/*.test.ts"]
 }

